WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,102 @@
 <% provide(:title, 'Competitions') %>
 
 <div class="container">
-  <h1><%= yield(:title) %></h1>
-
-  <%= link_to "New competition", new_competition_path %>
-  <table class="table">
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <div id="scroller-anchor"></div>
+  <div id="scroller">
+    <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+      <%= render 'index_form' %>
+    <% end %>
+  </div>
+    <% if @competitions.length == 0 %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <div class="row competition-list">
+          <div class="col-md-12">
+            <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+          </div>
+
+          <div class="col-md-6">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+          <div class="col-md-6">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions" %>
+          </div>
+        </div>
+      <% else %>
+        <div class="row competition-list">
+          <div class="col-xs-12 col-md-12">
+            <div id="map"></div>
```

Comment:
I'd spell out `competitions-map`, but do whatever you like here, just something more explicit than map, please =)